src/tests/ftest/io/nvme_pool_capacity.py

```diff
@@ -0,0 +1,268 @@
+                nvme_size (int): NVME size used in the testing
+            Returns:
+                None
+        """
+        pool = {}
+        cont = {}
+
+        for loop_count in range(0, total_count):
+            self.log.info("Running test %s", loop_count)
+            for val in range(0, num_pool):
+                pool[val] = TestPool(self.context,
+                                     dmg_command=self.get_dmg_command())
+                pool[val].get_params(self)
+                # Split total SCM and NVME size for creating multiple pools.
+                temp = int(float(scm_size) / num_pool)
+                pool[val].scm_size.update(str(temp))
+                temp = int(float(nvme_size) / num_pool)
+                pool[val].nvme_size.update(str(temp))
+                pool[val].create()
+                self.pool = pool[val]
+                display_string = "pool{} space at the Beginning".format(val)
+                self.pool.display_pool_daos_space(display_string)
+                nvme_size_begin = self.pool.get_pool_free_space("NVME")
+                for cont_val in range(0, num_cont):
+                    cont[cont_val] = TestContainer(pool[val])
+            m_leak = 0
+            for val in range(0, num_pool):
+                display_string = "Pool{} space at the End".format(val)
+                self.pool = pool[val]
+                self.pool.display_pool_daos_space(display_string)
+                nvme_size_end = self.pool.get_pool_free_space("NVME")
+                pool[val].destroy()
+                if (nvme_size_begin != nvme_size_end) and (m_leak == 0):
+                    m_leak = val
+            # After destroying pools, check memory leak for each test loop.
+            if m_leak != 0:
+                self.fail("Memory leak : iteration {0} \n".format(m_leak))
+
+    def test_run(self, num_pool=1):
+        """
+        Method Descripton:
+            This method is called with different test_cases.
+            Args:
+               num_pool (int): Total pools for running a test.
+            Returns:
+               None
+        """
+        num_jobs = self.params.get("no_parallel_job", '/run/ior/*')
+        # Create a pool
+        pool = {}
+
+        # Iterate through IOR different ior test sequence
+        for oclass, api, test, flags in product(self.ior_daos_oclass,
+                                                self.ior_apis,
+                                                self.ior_test_sequence,
+                                                self.ior_flags):
+            # Create the IOR threads
+            threads = []
+            for val in range(0, num_pool):
+                pool[val] = TestPool(self.context,
+                                     dmg_command=self.get_dmg_command())
+                pool[val].get_params(self)
+                # Split total SCM and NVME size for creating multiple pools.
+                pool[val].scm_size.value = int(float(test[0])) / num_pool
```

Comment:
Why do we set type to float first, then int? Can't we just set it to int directly?

Reply:
You are correct... This is a carryover... Python3 doesn't support long... It looks like int handles both int and long with the latest python 2x and python 3 versions... I can fix it...